slasher/detection/proposals/detector.go

```diff
@@ -0,0 +1,56 @@
+package proposals
+
+import (
+	"bytes"
+	"context"
+
+	ethpb "github.com/prysmaticlabs/ethereumapis/eth/v1alpha1"
+	"github.com/prysmaticlabs/prysm/beacon-chain/core/helpers"
+	"github.com/prysmaticlabs/prysm/slasher/db"
+	status "github.com/prysmaticlabs/prysm/slasher/db/types"
+	"github.com/prysmaticlabs/prysm/slasher/detection/proposals/iface"
+	"go.opencensus.io/trace"
+)
+
+var _ = iface.ProposalsDetector(&ProposeDetector{})
```

Comment:
Can this assertion live in the test file?

Reply:
done